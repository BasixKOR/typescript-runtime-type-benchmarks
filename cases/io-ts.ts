import * as t from 'io-ts';
import { Case } from './abstract';

const dataType = t.type({
  number: t.Int,
  negNumber: t.number,
  maxNumber: t.number,
  string: t.string,
  longString: t.string,
  boolean: t.boolean,
  deeplyNested: t.type({
    foo: t.string,
    num: t.number,
    bool: t.boolean,
  }),
});

export class IoTsCase extends Case implements Case {
  name = 'io-ts';

  validate() {
    const { data } = this

    if (dataType.is(data)) {
      return data;
    }

    throw new Error('Invalid');
  }
}
