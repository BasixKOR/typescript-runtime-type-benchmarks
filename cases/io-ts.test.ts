import { caseIoTs } from './io-ts'
import { DATA } from '../data'

describe('io-ts', () => {
  it('should pass', () => {
    const res = caseIoTs(DATA)
    expect(res).toEqual(DATA)

    if (res) {
      expect(res.number).toEqual(DATA.number)
    }
  })

  it('should fail negative number test', () => {
    const req = Object.assign({}, DATA, { neg_number: 1 })
    expect(() => caseIoTs(req)).toThrowErrorMatchingSnapshot()
  })

  it('should fail min length test', () => {
    const req = Object.assign({}, DATA, { long_string: '' })
    expect(() => caseIoTs(req)).toThrowErrorMatchingSnapshot()
  })
})
