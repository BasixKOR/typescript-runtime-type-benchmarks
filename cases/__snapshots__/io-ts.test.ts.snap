// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`io-ts should fail min length test 1`] = `"Invalid"`;

exports[`io-ts should fail negative number test 1`] = `"Invalid"`;
